src/newgrf.cpp

```diff
@@ -6636,6 +6636,16 @@ bool GetGlobalVariable(byte param, uint32_t *value, const GRFFile *grffile)
 			*value = static_cast<int32_t>(TimerGameCalendar::year);
 			return true;
 
+		/* case 0x25: // Reserved: For vehicles, this contains the GRFID of the grf that contains the corresponding Action3. */
+
+		case 0x26: // town cargo scale setting
+			*value = _settings_game.economy.town_cargo_scale_percentage;
+			return true;
+
+		case 0x27: // industry cargo scale setting
+			*value = _settings_game.economy.industry_cargo_scale_percentage;
+			return true;
```

Comment:
Why should this be exposed to NewGRFs? We are 'going behind their back' to change adjust production, but then let them know anyway?

Reply:
I agree, hide it from NewGRF.
Also: settings should go into GetPatchVariable(). GetGlobalVariable() is for dynamic values.

Reply:
As an aside, putting it into GetPatchVariable would mean that you would not be able to modify the value of the settings during multiplayer games at all, because of the use of Action D.
There is otherwise no reason for the added settings to have the SF_NO_NETWORK flag.
Changing the value during a single-player game would result in it not being visible to GRFs until the next save/load or GRF reload.
This would not be ideal, given that it's quite typical that you'll want to adjust the scaling as the game develops (usually, reduce it further as the network size increases).

If these variables were exposed to GRFs, GetGlobalVariable would be more useful than GetPatchVariable.

Reply:
I decided to not expose it to GRF. The original intention was to allow industry sets like FIRS to provide accurate helptext, but I agree that the potential for misuse might outweigh this.